src/v/cluster/types.cc

```diff
@@ -324,6 +359,11 @@ adl<cluster::topic_configuration>::from(iobuf_parser& in) {
     cfg.properties.retention_duration
       = adl<tristate<std::chrono::milliseconds>>{}.from(in);
 
+    if (version < 0) {
```

Comment:
Regarding usage of the serde library: IIUC having all fields inheriting from `serde::envelope` is not required - you can just define the `serde_write`/`serde_read` methods manually. I think this is pretty tempting - at least having this structure in the envelope will allow us to add fields in the future without any compatibility problems.

Reply:
done